parcels/field.py

```diff
@@ -1283,66 +1310,23 @@ def spatial_slip_interpolation(self, ti, z, y, x, time, particle=None, applyConv
             return u, v
 
     def eval(self, time, z, y, x, particle=None, applyConversion=True):
+        if self.U.interp_method in ["partialslip", "freeslip"]:
+            return self.slip_interpolation(time, z, y, x, particle=particle, applyConversion=applyConversion)
+
         if self.U.interp_method not in ["cgrid_velocity", "partialslip", "freeslip"]:
             u = self.U.eval(time, z, y, x, particle=particle, applyConversion=False)
             v = self.V.eval(time, z, y, x, particle=particle, applyConversion=False)
             if applyConversion:
                 u = self.U.units.to_target(u, z, y, x)
                 v = self.V.units.to_target(v, z, y, x)
```

Comment:
Could you elaborate on the deletion of the else block here?

Reply:
The 2D vs 3D is now dealt with in the interpolation methods themselves, so is not needed here anymore